Add references for CSS2/selectors

csnardi opened the pull request:
Many CSS2/selectors tests were missing references. This change adds in references for a large number of these tests.

The only major change to a test was in first-child-selector-003.xht; I don't see how it was correct as written (I flipped the colors).

<!-- Reviewable:start -->

<!-- Reviewable:end -->

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22746)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/328267361)

gsnedders commented on css/CSS2/selectors/first-child-selector-003.xht:
```diff
@@ -5,12 +5,13 @@
-   html { color: green; }
-   html:first-child { color: red ! important; }
+   html { color: red; }
+   html:first-child { color: green ! important; }
```
Hmm, I wonder if the intention here was `html > *:first-child`?

gsnedders replied:
(Also I don't understand why the `!important` is there.)

csnardi replied:
I'm not sure, going back to [the source](http://www.hixie.ch/tests/adhoc/css/selectors/child/001.xml) it's originally `root:first-child`.

gsnedders replied:
Let's assume it's a test that the root element matches `:first-child` and drop the `!important`?

csnardi replied:
Okay, I changed it to that.

csnardi committed:
Add references for CSS2/selectors

Many CSS2/selectors tests were missing references. This change adds in references for a large number of these tests.
